test/sequential/rolldown.updateL1FromL1.test.ts

```diff
@@ -634,12 +624,21 @@ describe("updateL1FromL1 - cancelResolution and deposit errors", () => {
       .on(chain)
       .buildUnsafe();
     await waitNewBlock();
-    await Rolldown.waitForReadRights(sequencer.keyRingPair.address, 50, chain);
-    const event2 = await Sudo.batchAsSudoFinalized(
+    await Rolldown.waitForReadRights(
+      sequencer.keyRingPair.address,
+      waitingPeriod,
+      chain,
+    );
+    await Sudo.batchAsSudoFinalized(
       Sudo.sudoAsWithAddressString(sequencer.keyRingPair.address, update2),
     );
-    const eventFiltered = filterAndStringifyFirstEvent(event2, "L1ReadStored");
-    await Rolldown.waitForL2UpdateExecuted(stringToBN(eventFiltered.range.end));
+    const event2 = await waitForEvents(
+      api,
+      "rolldown.RequestProcessedOnL2",
+      (await Rolldown.disputePeriodLength()).toNumber() * 4,
```

Comment:
what is that *4 about?

Reply:
We needed to replace the number 40 with a number tied to some constant, I chose disputePeriodLength